### ISSUE TITLE
CRT TLS context

### ISSUE THREAD

**Issue by aceeric** [awslabs/aws-crt-java#432]
Problem: The current CRT class `S3Client` does not support HTTP access to an S3 bucket. The code always establishes a TLS context - either by using the one provided to it, or by creating a default one. This then later results in the selection of HTTPS over 443. We want to use the AWS SDK + CRT in a Kubernetes service-meshed environment in which the endpoints are HTTP, and the mesh handles TLS.

Specifically:

In the CRT code, the code in `src/native/s3_client.c` in the function `JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_s3_S3Client_s3ClientNew` has code:
```
    ...
    struct aws_s3_client_config client_config = {
        .max_active_connections_override = max_connections,
        .region = region,
        .client_bootstrap = client_bootstrap,
        .tls_connection_options = tls_options,
        .signing_config = &signing_config,
        .part_size = (size_t)part_size,
        .throughput_target_gbps = throughput_target_gbps,
        .retry_strategy = retry_strategy,
        .shutdown_callback = s_on_s3_client_shutdown_complete_callback,
        .shutdown_callback_user_data = callback_data,
        .compute_content_md5 = compute_content_md5 ? AWS_MR_CONTENT_MD5_ENABLED : AWS_MR_CONTENT_MD5_DISABLED,
    };
    ....
```
In the `aws_s3_client_config` struct there is a member `enum aws_s3_meta_request_tls_mode tls_mode;` which is *not initialized* in the above block. Therefore, by default, this struct member has value zero which maps to enum `AWS_MR_TLS_ENABLED`. This enum is referenced in `source/s3_client.c` and has the effect of *always* enabling TLS.

In order to change this to support HTTP or HTTPS, I propose to submit a PR as follows:

1) Add a parameter `boolean tlsEnabled` to the JNI method `s3ClientNew` signature
2) Inside the JNI C code for that function, initialize the aforementioned struct member to enable or disable TLS on the basis of the parameter value. The code in `source/s3_client.c` already anticipates this so - no changes are required there
3) Modify `S3ClientOptions` and `S3Client` to also accommodate `tlsEnabled`

So this  change is localized in the CRT Java/JNI code, and doesn't touch any of the git sub-modules. We've already proofed this in-house. Does the idea of this PR seem acceptable? If so I will submit for review. Thanks.

**Comment by JonathanHenson** [comment]
Yes this is reasonable. Thanks for letting us know. I'm happy to review a PR.

**Comment by ron1** [comment]
Would it be better to include the endpoint as a parameter and enable tls only if the endpoint protocol is https?

**Comment by TingDaoK** [comment]
PR merged and new release published with the fix
